source/EthernetInterface.cpp

```diff
@@ -117,8 +122,20 @@ int EthernetInterface::connect(unsigned int timeout_ms) {
         netif_set_up(&netif);
         while (link_up == 0);
     }
+    _timeout.detach();
 
-    return inited;
+#define min(A,B) \
+    ((A) < (B) ? (A) : (B))
+#define max(A,B) \
+    ((A) > (B) ? (A) : (B))
+
+    if (min(link_up, if_up) < 0) {
+        return min(link_up, if_up);
+    } else if ((link_up == 0) && (if_up == 0)) {
+        return 0;
+    } else {
+        return max(link_up, if_up);
```

Comment:
According to EthernetInterface.h, this function is supposed to return 0 or a negative value:

>   * \return 0 on success, a negative number on failure

Not sure what you're returning here.